api/bazel/api_build_system.bzl

```diff
@@ -131,6 +131,14 @@ def api_proto_library(name, visibility = ["//visibility:private"], srcs = [], de
         ],
         visibility = ["//visibility:public"],
     )
+
+    native.cc_proto_library(
```

Comment:
So, without the PGV validation, there are two things we lose:
1. The ability to reject configurations that are somewhat nonsensical.
2. The guarantee that various preconditions on Envoy config processing code are met.

I think (1) is fine for an interim solution, but (2) is positively dangerous, since configuration can easily crash or pose security hazards to an Envoy instance. E.g. the config processing code might access an element in an array that has not been allocated, because it believes that due to PGV validation that the array is at least N elements long. If we want to accept (2) and the non-PGV variant for Windows, we probably want to slap a bunch of TODOs and warnings, together with tracking bugs, to ensure we bridge this gap.

Reply:
When building with the PGV validation, we ran into a few issues:

1. rules_go wasn't properly handling Windows paths. This appears to be fixed in 1c7fd7e58310b832decc04963ec13db204f79150, but this commit has not been released
2. in PGV, `_go_proto_library_gen_impl`: https://github.com/lyft/protoc-gen-validate/blob/master/bazel/go_proto_library.bzl#L130 was not generating an OS agnostic script
3. The `cc_proto_library`  rule from `@com_google_protobuf//:protobuf.bzl` did not de-duplicate command line arguments, causing it to hit the 32,768 character command line limit. The native Bazel `cc_proto_library` rule does not have this issue.

So it looks like the way forward would be something like:
1. Update project to latest rules_go once it is released
2. Fix up PGV `_go_proto_library_gen_impl` so it generates valid scripts for both Linux and Windows
3. PGV uses native Bazel `cc_proto_library` rules.

From the comment in `api/bazel/api_build_system.bzl`, it looks like the third step might not be possible at the moment

Reply:
Yeah, I think there's going to be some non-trivial changes to get us to native `cc_proto_library`. I don't think we can do release binaries for Windows or consider Windows supported beyond experimental until this is fixed though. We should add TODOs and tracking issues for now.